app/models/host/managed.rb

```diff
@@ -341,11 +342,14 @@ def compute?
     def compute_provides?(attr)
       false
     end
+
+    def without_orchestration
+      yield
+    end
   end
 
   before_validation :set_hostgroup_defaults, :set_ip_address
   after_validation :ensure_associations
-  before_validation :set_certname, :if => proc { |h| h.managed? && Setting[:use_uuid_for_certificates] } if SETTINGS[:unattended]
```

Comment:
Moved this callback in with the rest of the validations that only run when unattended is true